### PR TITLE
Sketch the UI interaction

### LINKED ISSUE #4: Sketch the UI page/representation interaction
Sketch the functionality for the application by:
- Defining the page/resource URIs
- Defining the representation obtained on a GET for each URI
- Defining the non-`GET` interactions.

### PR DESCRIPTION
**Pull request by TiagoMMDavid**
This PR sketches the web app routes
Closes GH-4

### CODE REVIEW

**Review by PTKay — APPROVED** [review]

**Review by dvsshadow — APPROVED** [review]

File: docs/routes/sketch.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  - Button to create a new classroom (`/orgs/{org-name}/classrooms/create`)
+
+- `/orgs/{org-name}/classrooms/{cid}`
+  - Information about the classroom (e.g. name, description)
+  - List with classroom assignments
+    - Each item has the assignment name, the due date and a link to `/orgs/{org-name}/classrooms/{cid}/assignments/{aid}`
+    - Each item has information about whether its an individual or group assignment
+    - Each item may show the number of students that finished the assignment (what if the assignment has multiple deliveries?)
+  - Button to create a new assignment (`/orgs/{org-name}/classrooms/{cid}/assignments/create`)
+
```
Comment (pmhsfelix) [inline docs/routes/sketch.md]: Suggestion: adds resources to represent the paginated list of assignments of a classroom.

File: docs/routes/sketch.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- `/orgs/{org-name}/classrooms/create`
+  - Supports POST method
+  - Form to create a new classroom
+  - A classroom has:
+    - Name
+    - Description (optional)
+    - ...
+
+- `/orgs/{org-name}/classrooms/{cid}/edit`
```
Comment (pmhsfelix) [inline docs/routes/sketch.md]: Avoid `edit` in the URI

File: docs/routes/sketch.md (hunk 1 above)
Comment (pmhsfelix) [reply docs/routes/sketch.md]: Same thing for all lists, i.e., have an independent resource to represent a paginated list of items.

**Review by dvsshadow — APPROVED** [review]

**Review by PTKay — APPROVED** [review]

**Review by dvsshadow — APPROVED** [review]

File: docs/sketches/interaction-sketch.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- Don't discriminate authenticated user (both students and teachers can log in).
+
+## Protecting Classroom Invites
+
+#### How to associate a CodeGarten account with an Institutional ID
+- The user inserts his institutional e-mail and ID and will receive a verification link on it, which will take him back to CodeGarten, verifying the ownership of the institutional e-mail
+
+#### How do Classroom Invites work
+- Teacher generates an invite link for the classroom
+- Students, via that link, will have to log in with their CodeGarten account (create one if it doesn't exist), and link their institutional e-mail with their CodeGarten account if needed (process described above)
```
Comment (pmhsfelix) [inline docs/sketches/interaction-sketch.md:12]: I would avoid having to manage user's credentials. Instead, we can delegate authentication to GitHub.

File: docs/sketches/interaction-sketch.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+#### Institutions:
+- Have a name and a unique ID
+- Have a way to associate a student number to a student e-mail (for instance, type an email with a {id} where the number would be placed)
+- Used when inviting students
+- Exclusive to the logged in user? Shared with everyone?
```
Comment (pmhsfelix) [inline docs/sketches/interaction-sketch.md:18]: I don't understand this question.

File: docs/sketches/routes/assignments.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+## /orgs/{oid}/classrooms/{cid}/assignments (supports POST to create)
+  - Accepts a query parameter to specify the current page 
+  - Paginated list with all student assignments
+    - Each item has the assignment name, the due date and a link to `/orgs/{oid}/classrooms/{cid}/assignments/{aid}`
+    - Each item has information about whether its an individual or group assignment
+    - Each item may show the number of students that finished the assignment (what if the assignment has multiple deliveries?)
+  - Button to create a new assignment (`/orgs/{oid}/classrooms/{cid}/assignments/new`)
+
+## /orgs/{oid}/classrooms/{cid}/assignments/new
```
Comment (pmhsfelix) [inline docs/sketches/routes/assignments.md:24]: Do we really need this extra route ending in `new`?

File: docs/sketches/routes/classrooms.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+## /orgs/{oid}/classrooms (supports POST to create)
+  - Accepts a query parameter to specify the current page 
+  - Paginated list with all classrooms
+    - Each item has the classroom name and a link to `/orgs/{oid}/classrooms/{cid}`
+    - Each item also has the number of students, and a list of the most recent assignments
+    - Each item has a small icon to indicate if the user is a student or a teacher
+  - Button to create a new classroom (`/orgs/{oid}/classrooms/new`)
+  - If the user is the organization owner, button to create a new classroom (`/orgs/{oid}/classrooms/new`)
+
+## /orgs/{oid}/classrooms/new
```
Comment (pmhsfelix) [inline docs/sketches/routes/classrooms.md:19]: Same

**Review by pmhsfelix — APPROVED** [review]
Left some comment, but I think it is in a good state for we to begin development.

**Pull request merged**